bundles/automation/org.eclipse.smarthome.automation.module.core/src/main/java/org/eclipse/smarthome/automation/module/core/handler/GenericEventTriggerHandler.java

```diff
@@ -92,6 +92,7 @@ public void receive(Event event) {
         if (callback != null) {
             logger.trace("Received Event: Source: " + event.getSource() + " Topic: " + event.getTopic() + " Type: "
                     + event.getType() + " Payload: " + event.getPayload());
+            // TODO MKA: Why is the topic checked against the source and not event.getSource()?
```

Comment:
simply because for some event types the event source is not the item (or even not set...)